docs/python.md

```diff
@@ -22,23 +36,25 @@ class Predictor(BasePredictor):
         return output
 ```
 
-Tip: Run [`cog init`](getting-started-own-model#initialization) to generate an annotated `predict.py` file that can be used as a starting point for setting up your model.
-
-You need to override two functions: `setup()` and `predict()`.
+Your Predictor class should define two methods: `setup()` and `predict()`.
 
 ### `Predictor.setup()`
 
-Set up the model for prediction so multiple predictions run efficiently. Include any expensive one-off operations in here like loading trained models, instantiate data transformations, etc.
+Prepare the model so multiple predictions run efficiently.
+
+Use this _optional_ method to include any expensive one-off operations in here like loading trained models, instantiate data transformations, etc.
 
 It's best not to download model weights or any other files in this function. You should bake these into the image when you build it. This means your model doesn't depend on any other system being available and accessible. It also means the Docker image ID becomes an immutable identifier for the precise model you're running, instead of the combination of the image ID and whatever files it might have downloaded.
 
 ### `Predictor.predict(**kwargs)`
 
-Run a single prediction. This is where you call the model that was loaded during `setup()`, but you may also want to add pre- and post-processing code here.
+Run a single prediction.
+
+This _required_ method is where you call the model that was loaded during `setup()`, but you may also want to add pre- and post-processing code here.
 
-The `predict()` function takes an arbitrary list of named arguments, where each argument name must correspond to a `@cog.input()` annotation.
+The `predict()` method takes an arbitrary list of named arguments, where each argument name must correspond to an [`Input()`](#inputkwargs) annotation.
 
-`predict()` can output strings, numbers, `pathlib.Path` objects, or lists or dicts of those types. We are working on support for other types of output, but for now we recommend using base-64 encoded strings or `pathlib.Path`s for more complex outputs.
+`predict()` can return strings, numbers, `pathlib.Path` objects, or lists or dicts of those types. You can also define a custom [`Output()`](#outputbasemodel) for more complex return types.
 
 #### Returning `pathlib.Path` objects
```

Comment:
We aren't returning `pathlib.Path` objects any longer -- we're returning `cog.Path` (or `cog.File`) objects.

A better heading for this section would perhaps be "Returning files" or "Outputting files".